### ISSUE TITLE
[FR] support search URLs

### ISSUE THREAD

**Issue by ildar** [TeamNewPipe/NewPipe#2313]
opening from external apps. Think URL intents should be added and such...

- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I checked if the issue/feature exists in the latest version.
- [ ] I did use the [incredible bugreport to markdown converter](https://teamnewpipe.github.io/CrashReportToMarkdown/) to paste bug reports.

**Comment by haux205** [comment]
I would like to work on this issue. I'm new to open source so can someone walk me through the process ?

**Comment by Stypox** [comment]
@haux205 yeah sure, you are welcome :-D

NewPipe uses a library called [NewPipeExtractor](https://github.com/TeamNewPipe/NewPipeExtractor/) to handle most service-specific things (such as parsing data). This also includes link handling, so you should change things there. So you probably have to take a look to [YouTube's link handlers](https://github.com/TeamNewPipe/NewPipeExtractor/tree/dev/extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/linkHandler), or more specifically, the [Youtube search query link handler](https://github.com/TeamNewPipe/NewPipeExtractor/tree/dev/extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/linkHandler/YoutubeSearchQueryHandlerFactory.java) (which currently is only able to generate urls).
Since no service so far ever needed to handle search urls you need to add the check in [the function that handles all urls](https://github.com/TeamNewPipe/NewPipeExtractor/blob/dev/extractor/src/main/java/org/schabi/newpipe/extractor/StreamingService.java#L272) along with a new search link type.

Since the urls you will add have the same domain as normal youtube urls, you shouldn't need to change anything in the app's Android manifest. Handle search urls here in the in the [MainActivity intent handler](https://github.com/TeamNewPipe/NewPipe/blob/dev/app/src/main/java/org/schabi/newpipe/MainActivity.java#L635-L677).

If you have any more questions or need help with setting up the environment, feel free to ask ;-)

**Comment by B0pol** [comment]
https://github.com/TeamNewPipe/NewPipeExtractor/pull/124 is already working on search filters

**Comment by SydneyDrone** [comment]
I'm interested in fixing this issue, but I don't know the PR mentioned above has fixed this problem or not. Another question is that 
I'm not sure the idea of fixing this issue is still the same.

**Comment by Stypox** [comment]
@SydneyDrone no, the PR mentioned above was closed. This should be solvable without too many issues, just follow the points in my comment above. Thanks :-)

**Comment by Sola0404** [comment]
Hi, I am a computer science student in ANU, and I really hope I could try to fix this issue and create PR for it. Could you please assign me to this issue? Thanks in advance!
